src/main/java/ru/test/restservice/api/CompilerController.java

```diff
@@ -23,12 +22,13 @@
     /**
      * Метод, обрабатывающий запросы на компиляцию с фронтенда
      *
-     * @param target имя компилируемого .tex файла
+     * @param targetFilepath путь до компилируемого .tex файла
      * @return JSON с ошибкой компиляции или путь к pdf
      */
     @PostMapping("/compile")
-    public CompilationResultDTO compileHandler(@RequestParam String target, @RequestBody List<FileItemDTO> files) throws IOException {
+    public CompilationResultDTO compileHandler(@RequestParam String targetFilepath, @RequestBody List<FileItemDTO> files) throws IOException {
         fileService.rewriteFiles(files);
-        return compilerService.compileTexFile(compilerService.workingFolder, target);
+        FileItemDTO file = files.stream().filter(f -> targetFilepath.equals(f.path)).findFirst().get();
```

Comment:
такое не стоит писать в одну строчку
```suggestion
        FileItemDTO file = files.stream()
                .filter(f -> targetFilepath.equals(f.path))
                .findFirst().get();
```